megamek/src/megamek/server/Server.java

```diff
@@ -30588,35 +29482,26 @@ public void addSmoke(ArrayList<Coords> coords, int windDir, boolean bInferno) {
             // Have to check if it's inferno smoke or from a heavy/hardened
             // building
             // - heavy smoke from those
-            if (bInferno
-                || (Building.MEDIUM < smokeHex
-                    .terrainLevel(Terrains.FUEL_TANK))
-                || (Building.MEDIUM < smokeHex
-                    .terrainLevel(Terrains.BUILDING))) {
-                if (smokeHex.terrainLevel(Terrains.SMOKE)
-                    == SmokeCloud.SMOKE_HEAVY) {
+            if (bInferno || (Building.MEDIUM < smokeHex.terrainLevel(Terrains.FUEL_TANK))
+                    || (Building.MEDIUM < smokeHex.terrainLevel(Terrains.BUILDING))) {
+                if (smokeHex.terrainLevel(Terrains.SMOKE) == SmokeCloud.SMOKE_HEAVY) {
                     // heavy smoke fills hex
                     r = new Report(5180, Report.PUBLIC);
-                    r.add(smokeCoords.getBoardNum());
-                    addReport(r);
                     smokeLevel = SmokeCloud.SMOKE_HEAVY;
                 } else {
-                    smokeLevel = Math.max(smokeLevel, SmokeCloud.SMOKE_HEAVY);
                     r = new Report(5185, Report.PUBLIC);
-                    r.add(smokeCoords.getBoardNum());
-                    addReport(r);
-                    smokeLevel = SmokeCloud.SMOKE_HEAVY;
+                    smokeLevel = Math.max(smokeLevel, SmokeCloud.SMOKE_HEAVY); // TODO : Windchild verify this change is correct
```

Comment:
We were doing some math then overwriting it. Should be the math.max, or the SmokeCloud.SMOKE_HEAVY?

Reply:
I don't know what the Math.max line is there for. We get to this block if there is a fire that generates heavy smoke and there is not already heavy smoke in that hex. So `smokeLevel = SmokeCloud.SMOKE_HEAVY` is correct.